openssl/src/x509/mod.rs

```diff
@@ -349,6 +350,61 @@ impl X509Generator {
         }
     }
 
+    /// Sets the certificate public-key, then signs it by the CA key and certificate and return it
+    /// Note: That the bit-length of the private key is used (set_bitlength is ignored)
+    pub fn sign_by_ca(&self, subject_pkey: &PKey, ca_cert: &X509, ca_pkey: &PKey) -> Result<X509, ErrorStack> {
```

Comment:
Would it be possible to reduce the code duplication in the internals of `sign` and `sign_by_ca`?